Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1313 @@
+                {
+                    bindPreference = addr.Value;
+                    break;
+                }
+                else if ((addr.Key.Address.Equals(IPAddress.Any) || addr.Key.Address.Equals(IPAddress.IPv6Any)) && isInExternalSubnet)
+                {
+                    // External.
+                    bindPreference = addr.Value;
+                    break;
+                }
+                else if (addr.Key.Contains(source))
+                {
+                    // Match ip address.
+                    bindPreference = addr.Value;
+                    break;
+                }
+            }
+
+            if (string.IsNullOrEmpty(bindPreference))
+            {
+                return false;
+            }
+
+            // Has it got a port defined?
+            var parts = bindPreference.Split(':');
+            if (parts.Length > 1)
+            {
+                if (int.TryParse(parts[1], out int p))
+                {
+                    bindPreference = parts[0];
+                    port = p;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Attempts to match the source against a user defined bind interface.
+        /// </summary>
+        /// <param name="source">IP source address to use.</param>
+        /// <param name="isInExternalSubnet">True if the source is in the external subnet.</param>
+        /// <param name="result">The result, if a match is found.</param>
+        /// <returns><c>true</c> if a match is found, <c>false</c> otherwise.</returns>
+        private bool MatchesBindInterface(IPObject source, bool isInExternalSubnet, out string result)
+        {
+            result = string.Empty;
+            var addresses = _bindAddresses.Exclude(_bindExclusions);
+
+            int count = addresses.Count;
+            if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses[0].Equals(IPAddress.IPv6Any)))
+            {
+                // Ignore IPAny addresses.
+                count = 0;
+            }
+
+            if (count != 0)
+            {
+                // Check to see if any of the bind interfaces are in the same subnet.
+
+                IPAddress? defaultGateway = null;
+                IPAddress? bindAddress = null;
+
+                if (isInExternalSubnet)
+                {
+                    // Find all external bind addresses. Store the default gateway, but check to see if there is a better match first.
+                    foreach (var addr in addresses.OrderBy(p => p.Tag))
+                    {
+                        if (defaultGateway == null && !IsInLocalNetwork(addr))
+                        {
+                            defaultGateway = addr.Address;
+                        }
+
+                        if (bindAddress == null && addr.Contains(source))
+                        {
+                            bindAddress = addr.Address;
+                        }
```

Comment:
```suggestion
                        }
                        
                        if (defaultGateway != null && bindAddress != null)
                        {
                            break;
                        }
```

Reply:
No need to keep iterating if it's done

Reply:
did wonder at the time - then thought about the number of IP addresses users would have. Would it cause more code or less code to be run with this checked on every iteration???

Have added it anyway.

Reply:
LOC is not a very useful metric. The server code should be concise and performant. The latter sometimes requires a few sacrifices.